frontend/lib/types/DomainType.cpp

```diff
@@ -29,6 +30,12 @@
 namespace chpl {
 namespace types {
 
+const ID DomainType::rankId = ID(UniqueString(), 0, 0);
+const ID DomainType::rectangularIdxTypeId = ID(UniqueString(), 1, 0);
+const ID DomainType::nonRectangularIdxTypeId = ID(UniqueString(), 0, 0);
+const ID DomainType::stridesId = ID(UniqueString(), 2, 0);
+const ID DomainType::parSafeId = ID(UniqueString(), 1, 0);
+const ID DomainType::runtimeTypeId = ID(UniqueString(), 3, 0);
```

Comment:
Need to update `getRectangularType` and `getAssociativeType` to use these new constants for their substitution IDs